sphinx/source/docs/user_guide/data.rst

```diff
@@ -0,0 +1,363 @@
+
+    output_file("group_filter.html")
+
+    source = ColumnDataSource(flowers)
+    view1 = CDSView(source=source, filters=[GroupFilter(column_name='species', group='versicolor')])
+
+    plot_size_and_tools = {'plot_height': 300, 'plot_width': 300,
+                            'tools':['box_select', 'reset', 'help']}
+
+    p1 = figure(title="Full data set", **plot_size_and_tools)
+    p1.circle(x='petal_length', y='petal_width', source=source, color='black')
+
+    p2 = figure(title="Setosa only", x_range=p1.x_range, y_range=p1.y_range, **plot_size_and_tools)
+    p2.circle(x='petal_length', y='petal_width', source=source, view=view1, color='red')
+
+    show(gridplot([[p1, p2]]))
+
+CustomJSFilter
+~~~~~~~~~~~~~~
+
+You can also create a filter with custom functionality. This type of filter, the
+|CustomJSFilter|, can be written in three ways, using Javascript, Coffeescript, or Python.
+For all three, the custom code needs to return either a list of indices or a list of
+booleans that represents the filtered subset. The |ColumnDataSource| that is associated
+with the |CDSView| this filter is added to will be available at render time with the
+variable ``source``.
+
+Javascript
+''''''''''
+
+To create a |CustomJSFilter| with custom functionality written in JavaScript,
+pass in the JavaScript code as a string to the parameter ``code``:
+
+.. code-block:: python
+
+    custom_filter = CustomJSFilter(code='''
+    var indices = [];
+
+    // iterate through rows of data source and see if each satisfies some constraint
+    for (var i = 0; i <= source.get_length(); i++){
+        if (source.data['some_column'][i] == 'some_value'){
+            indices.push(true);
+        } else {
+            indices.push(false);
+        }
+    }
+    return indices;
+    ''')
+
+Coffeescript
+''''''''''''
+
+You can also write code for the ``CustomJSFilter`` in `CoffeeScript`_, and
+use the ``from_coffeescript`` class method, which accepts the ``code`` parameter:
+
+.. code-block:: python
+
+    custom_filter_coffee = CustomJSFilter.from_coffeescript(code='''
+    z = source.data['z']
+    indices = (i for i in [0...source.get_length()] when z[i] == 'b')
+    return indices
+    ''')
+
+Python
+''''''
+
+A CustomJSFilter can also be created from a Python function. To use this functionality
+you need the Flexx library (install with ``conda install -c bokeh flexx`` or
+``pip install flexx``). It is also important to note that not all Python code is
+supported, only the subset of Python that can be translated to Javascript using PyScript.
+
+.. code-block:: python
+
+    def custom_python_filter():
+        z = source.data['z']
+        indices = [True if z[i] == 'c' else False for i in range(len(z)) ]
+        return indices
+
+    custom_filter_pyscript = CustomJSFilter.from_py_func(custom_python_filter)
+
+For more information about the subset of Python that is supported,
+see the `PyScript documentation`_.
+
+.. _userguide_data_linked_selection:
```

Comment:
I'd just as soon leave this section out. I'm not sure I think this kind of transpilation is a reasonable thing anymore.